### PR TITLE
HLM-5143 Adding helm chart changes for plan-service

### PR DESCRIPTION
**Pull request by Priyanka-eGov**

### CODE REVIEW

File: deploy-as-code/helm/charts/health-services/plan-service/values.yaml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+replicas: "1"
+healthChecks:
+  enabled: true
+  livenessProbePath: "/plan-service/health"
+  readinessProbePath: "/plan-service/health"
+appType: "java-spring"
+tracing-enabled: true
+heap: "-Xmx192m -Xms192m"
+java-args: ""
+persist-yml-path: "file:///work-dir/health-campaign-config/egov-persister/plan-service-persister.yml"
```
Comment (shashwat-egov) [inline deploy-as-code/helm/charts/health-services/plan-service/values.yaml:35]: why is this required?

File: deploy-as-code/helm/charts/health-services/plan-service/values.yaml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      configMapKeyRef:
+        name: egov-config
+        key: tracer-errors-provideexceptionindetails
+  {{- end }}
+  - name: STATE_LEVEL_TENANT_ID
+    valueFrom:
+      configMapKeyRef:
+        name: egov-config
+        key: egov-state-level-tenant-id 
+  - name: EGOV_IDGEN_HOST
```
Comment (shashwat-egov) [inline deploy-as-code/helm/charts/health-services/plan-service/values.yaml:56]: why is ID gen related properties required?

**Review by shashwat-egov — CHANGES_REQUESTED** [review]

File: deploy-as-code/helm/charts/health-services/plan-service/values.yaml (hunk 1 above)
Comment (Priyanka-eGov) [reply deploy-as-code/helm/charts/health-services/plan-service/values.yaml]: not required, have removed it

File: deploy-as-code/helm/charts/health-services/plan-service/values.yaml (hunk 2 above)
Comment (Priyanka-eGov) [reply deploy-as-code/helm/charts/health-services/plan-service/values.yaml]: have removed them.

File: deploy-as-code/helm/charts/health-services/plan-service/values.yaml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  - name: EGOV_MDMS_HOST
+    valueFrom:
+      configMapKeyRef:
+        name: egov-service-host
+        key: {{ index .Values "egov-mdms-service-host-key" | quote }}
+  - name: EGOV_MDMS_SEARCH_ENDPOINT
+    value: "/egov-mdms-service/v1/_search"
+  - name: EGOV_MDMS_INTEGRATION_ENABLED
+    value: "true"
+  - name: EGOV_LOCALIZATION_HOST
```
Comment (shashwat-egov) [inline deploy-as-code/helm/charts/health-services/plan-service/values.yaml:64]: why is localization host needed?

File: deploy-as-code/helm/charts/health-services/plan-service/values.yaml [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+egov-mdms-service-host-key: "egov-mdms-service"
+
+# Additional Container Envs
+env: |
+  - name: SPRING_KAFKA_CONSUMER_GROUP_ID
+    value: plan-service
+  - name: SPRING_KAFKA_PRODUCER_KEY_SERIALIZER
+    value: org.apache.kafka.common.serialization.StringSerializer
+  {{- if index .Values "global" "tracer-errors-provideexceptionindetails" }}
+  - name: TRACER_ERRORS_PROVIDEEXCEPTIONINDETAILS
```
Comment (shashwat-egov) [inline deploy-as-code/helm/charts/health-services/plan-service/values.yaml:44]: what does this value correspond to?

**Review by shashwat-egov — CHANGES_REQUESTED** [review]

File: deploy-as-code/helm/charts/health-services/plan-service/values.yaml (hunk 4 above)
Comment (Priyanka-eGov) [reply deploy-as-code/helm/charts/health-services/plan-service/values.yaml:44]: its to decide whether to print the whole stack trace or not during exception handling.

**Review by shashwat-egov — APPROVED** [review]
